components/Form/Edit.js

```diff
@@ -0,0 +1,123 @@
+import { useState, useEffect } from "react";
+import { FormContainer, FormButton } from "./styles-form";
+import { Heading } from "./styles-form";
+import { useRouter } from "next/router";
+
+export default function EditRecipe() {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const initialIngredients = [{ name: "", quantity: "" }];
+
+  const [recipe, setRecipe] = useState({
+    name: "",
+    ingredients: initialIngredients,
+    quantity: 0,
+    steps: "",
+    timers: "",
+  });
+
+  useEffect(() => {
+    if (id) {
+      fetchRecipe(id);
+    }
+  }, [id]);
+
+  const fetchRecipe = async (id) => {
+    try {
+      const response = await fetch(`/api/recipes/${id}`);
+      const data = await response.json();
+      setRecipe(data);
+    } catch (error) {
+      console.error("Error fetching recipe:", error);
+    }
+  };
+
+  const handleEdit = async (event) => {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const updatedRecipe = Object.fromEntries(formData);
+
+    try {
+      const response = await fetch(`/api/recipes/${id}`, {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(updatedRecipe),
+      });
+      if (!response.ok) {
+        window.alert("Problem updating the recipe!");
+        return;
+      }
+
+      await fetchRecipe(id);
+
+      router.push("/");
+    } catch (error) {
+      window.alert("Problem updating the recipe!");
+    }
+  };
+
+  const handleInputChange = (event) => {
+    const { name, value } = event.target;
+    setRecipe((prevRecipe) => ({
+      ...prevRecipe,
+      [name]: value,
+    }));
+  };
+
+  const handleIngredientChange = (index, field, event) => {
+    const updatedIngredients = [...recipe.ingredients];
+    updatedIngredients[index][field] = event.target.value;
+
+    setRecipe({
+      ...recipe,
+      ingredients: updatedIngredients,
+    });
+  };
```

Comment:
I don't know why you have that much different handlers here. I think it is enough to have the handleSubmit and there get the formdata and fetch the api/recipes/${id}.